resources.Groups.department.ReferenceExample.ps1

```diff
@@ -24,102 +131,206 @@ catch {
+                $ADGroupName = ("$adGroupNamePrefix" + "$($resource.DisplayName)" + "$adGroupNameSuffix")
+                $ADGroupDescription = ("$adGroupDescriptionPrefix" + "$($resource.DisplayName)" + "$adGroupDescriptionSuffix")
 
-            # If resource does not exist
-            if ($groupExists -eq $False) {
-                <# Resource creation preview uses a timeout of 30 seconds
-                while actual run has timeout of 10 minutes #>
-                Write-Information "Creating $($distinguishedName)"
+                # Example: Title 
+                # $ADGroupName = ("$adGroupNamePrefix" + "$($resource.Name)" + "$adGroupNameSuffix")
+                # $ADGroupDescription = ("$adGroupDescriptionPrefix" + "$($resource.Name)" + "$adGroupDescriptionSuffix")
 
-                if (-Not($dryRun -eq $True)) {
-                    $NewADGroup = New-ADGroup @ADGroupParams
+                # Example: Custom object (custom object consists of all custom properties)
+                # $ADGroupName = ("$adGroupNamePrefix" + "$($resource.CustomField1)" + "$($resource.CustomField2)" + "$adGroupNameSuffix")
+                # $ADGroupDescription = ("$adGroupDescriptionPrefix" + "$($resource)" + "$adGroupDescriptionSuffix")
 
+                $ADGroupParams = @{
+                    Name           = $ADGroupName
+                    SamAccountName = $ADGroupName
+                    GroupCategory  = "Security"
+                    GroupScope     = "Global"
+                    DisplayName    = $ADGroupName
+                    Path           = $path
+                    Description    = $ADGroupDescription
+                }
+
+                $distinguishedName = "CN=$($ADGroupParams.SamAccountName),$($ADGroupParams.Path)"
+                $adGroup = $adGroupsGrouped["$($ADGroupParams.SamAccountName)"]
+                if ($null -ne $adGroup) {
+                    $groupExists = $true
+                }
+                else {
+                    $groupExists = $false
+                }
+                # If resource does not exist
+                if ($groupExists -eq $False) {
+                    <# Resource creation preview uses a timeout of 30 seconds
+                    while actual run has timeout of 10 minutes #>
+                    if (-Not($dryRun -eq $True)) {
+                        if ($debug -eq $true) {
+                            Write-Information "Debug: Creating group [$($distinguishedName)]"
+                            Write-Information "Debug: Group parameters: $($ADGroupParams | ConvertTo-Json)"
+                        }
+    
+                        $NewADGroup = New-ADGroup @ADGroupParams
+    
+                        $auditLogs.Add([PSCustomObject]@{
+                                Message = "Created group [$($distinguishedName)]"
+                                # Message = "Created resource for $($resource.name) - $distinguishedName"
```

Comment:
is dit een voorbeeld? Ik vind dit eigenlijk mooier :-)

Reply:
Is inderdaad een voorbeeld, aangezien de resource naam kan verschillen per item, is anders voor functie en afdeling, maar ook weer voor custom velden.
Ik kan wel het voorbeeld de standaard maken en evt. anderen als voorbeeld erbij zetten.